spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/BaseCatalog.java

```diff
@@ -45,4 +45,17 @@ public Procedure loadProcedure(Identifier ident) throws NoSuchProcedureException
 
     throw new NoSuchProcedureException(ident);
   }
+
+  /**
+   * Invalidate cached table metadata from current catalog.
+   * <p>
+   * If the table is already loaded or cached, drop cached data. If the table does not exist or is
+   * not cached, do nothing.
+   *
+   * @param ident a table identifier
+   * @return true if the table is cached, false otherwise
+   */
+  public boolean invalidateTableIfCached(Identifier ident) {
```

Comment:
I don't think that we need this method. We only need to check whether the Iceberg cache had the table. No need to change the Spark catalog methods.

Reply:
Oh, I see why you added this. I think it isn't worth this change. Let's just call both invalidate methods instead.